### PR TITLE
rgw: The error reported in the "rgw_s3_prepare_decrypt" function is unreasonable

### PR DESCRIPTION
**Pull request by ybwang0211**
When decrypting the data encrypted with the "SSE-KMS" method, the "InvalidAccessKeyId" error is reported when the KMS key is unavailable, which is unreasonable. Therefore, we added the "InvalidKmsKeyId" error code to correct this problem.

Fixes: https://tracker.ceph.com/issues/55799
Signed-off-by: wangyingbin <ybwang0211@163.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/rgw_common.cc [hunk 1]
```diff
@@ -85,6 +85,7 @@ rgw_http_errors rgw_http_s3_errors({
     { EPERM, {403, "AccessDenied" }},
     { ERR_SIGNATURE_NO_MATCH, {403, "SignatureDoesNotMatch" }},
     { ERR_INVALID_ACCESS_KEY, {403, "InvalidAccessKeyId" }},
+    { ERR_INVALID_KMS_KEY, {403, "InvalidKmsKeyId" }},
```
Comment (cbodley) [inline src/rgw/rgw_common.cc:88]: i don't see `InvalidKmsKeyId` listed in https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html, where does this come from?

i think we should just use `EINVAL` for these instead

**Comment by ybwang0211** [comment]
您好，这里是王迎彬的邮箱，已经收到您的邮件，谢谢。

File: src/rgw/rgw_common.cc (hunk 1 above)
Comment (ybwang0211) [reply src/rgw/rgw_common.cc]: The "InvalidKmsKeyId" I added is not in https://docs.aws.amazon.com/AmazonS3/latest/API/ErrorResponses.html, in this case EINVAL is more reasonable, so I have Modified him as you suggested, thanks.

**Review by cbodley — APPROVED** [review]

**Comment by ybwang0211** [comment]
您好，这里是王迎彬的邮箱，已经收到您的邮件，谢谢。

**Comment by cbodley** [comment]
passed qa in https://pulpito.ceph.com/cbodley-2022-08-18_23:33:15-rgw-wip-cbodley-testing-distro-default-smithi/

**Pull request merged**